app/graphql/types/interface/media.rb

```diff
@@ -101,6 +101,23 @@ def titles
     null: false,
     description: 'A large banner image for this media'
 
+  field :my_library_entry, Types::LibraryEntry,
+    null: true,
+    description: 'Your library entry related to this media.'
+
+  def my_library_entry
+    if context[:token].blank?
+      raise GraphQL::ExecutionError, 'You must be authorized to view your library entry'
+    end
+
+    RecordLoader.for(
+      LibraryEntry,
+      column: :user_id,
+      where: { media_id: object.id, media_type: type },
```

Comment:
Should this be `object.type` or does `type` work too?

Reply:
so `object.type` doesn't actually work because Anime/Manga doesn't have a `type`. There is custom code above in the `Media` interface
```
  # Types::Anime -> Anime
  def type
    self.class.name.split('::').last
  end
 ```
 which allows us to use our interface via copy/pasta in all the main types.